### PR TITLE
Edit variables

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

**Comment by valentinvieriu** [comment]
Please check https://github.com/kyma-project/kyma/issues/10311
I've added some ui changes in there.

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/LambdaVariables/VariableInputs/ResourceVariableInput.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+  }
+
+  const resourceOptions = (resources || []).map(({ metadata }) => ({
+    key: metadata.name,
+    text: metadata.name,
+  }));
+
+  let resourceKeysOptions = Object.keys(selectedResource.data || []).map(
+    key => {
+      return {
```
Comment (parostatkiem) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/LambdaVariables/VariableInputs/ResourceVariableInput.js:179]: You can use a direct object return as above

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/LambdaVariables/test/validation.test.js [hunk 2]
```diff
@@ -104,6 +107,57 @@ describe('validateVariable', () => {
...
+          key: null,
+        },
+      },
+      validation: VARIABLE_VALIDATION.NONE,
+    };
+
+    expect(validateVariable(variables, currentVariable)).toBeFalsy();
+  });
+
+  test('should return true if currentVariable valueFrom.name or key are present', () => {
```
Comment (parostatkiem) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/LambdaVariables/test/validation.test.js:144]:
```suggestion
  test('should return true if currentVariable valueFrom.name and key are present', () => {
```

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/LambdaVariables/validation.js [hunk 3]
```diff
@@ -7,23 +7,27 @@ import { CONFIG } from 'components/Lambdas/config';
...
-  });
+  return [...customVariables, ...customValueFromVariables].map(
+    (variable, _, array) => {
+      const validation = getValidationStatus({
+        userVariables: array,
+        injectedVariables,
+        restrictedVariables: CONFIG.restrictedVariables,
+        varName: variable.name,
+        varID: variable.id,
+        varValue: variable.value ? variable.value : variable.valueFrom,
```
Comment (parostatkiem) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/LambdaVariables/validation.js:21]:
```suggestion
        varValue: variable.value || variable.valueFrom,
```

**Review by parostatkiem — CHANGES_REQUESTED** [review]

**Review by parostatkiem — APPROVED** [review]

**Pull request merged**